File: app/src/main/java/com/loe/test/TestInterface.java
package com.loe.test;

import com.loe.http.Link;
import com.loe.http.annotation.DELETE;
import com.loe.http.annotation.GET;
import com.loe.http.annotation.IS_JSON;
import com.loe.http.annotation.NO_DEALER;
import com.loe.http.annotation.PUT;
import com.loe.http.annotation.ParamFile;
import com.loe.http.annotation.Header;
import com.loe.http.annotation.LINK;
import com.loe.http.annotation.POST;
import com.loe.http.annotation.Param;
import com.loe.http.annotation.TAG;

import java.io.File;

public interface TestInterface
{
    @DELETE("test/request.php")
    @IS_JSON
    Link go(@Param("名称") String name,
            @Header("some") int age);

    @POST("test/request.php")
    @NO_DEALER
    Link pf(@Param("ddd是") String name,
            @ParamFile("文件") File file);

    void back();
}

File: app/src/main/java/com/loe/test/TestInterface.kt
package com.loe.test

import com.loe.http.Link
import com.loe.http.annotation.*
import org.json.JSONObject
import java.io.File

interface TestInterface {
    @DELETE("test/request.php")
    @JSON
    @Headers(
        "kjs = jdhdssssssssssssssss123",
        "zzzzzlkjsj = hdhbbmxm33333333xn2222"
    )
    @Params(
        "kjs = jdhdssssssssssssssss123",
        "zzzzzlkjsj = hdhbbmxmxn22220000000000"
    )
    fun go(
        @Param("名称") name: String = "大幅度发到付",
        @Param("json测") jc: JSONObject = JSONObject()
            .put("", ""),
        @Header("some") age: Int = 9999999
    ): Link

    @POST("test/request.php")
    @NO_DEAL
    fun pf(
        @Param("ddd是") name: String,
        @ParamFile("文件") file: File
    ): Link

    fun back()
}